iothub/device/src/Microsoft.Azure.Devices.Client.csproj

```diff
@@ -50,13 +50,13 @@
 
   <!--Nuget package dependencies-->
   <ItemGroup>
-    <PackageReference Include="MQTTnet" Version="4.0.0.167" />
+    <PackageReference Include="MQTTnet" Version="4.1.2.350" />
     <PackageReference Include="Microsoft.Azure.Amqp" Version="2.5.12" />
     <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
-    <PackageReference Include="Microsoft.Win32.Registry" Version="4.5.0" />
+    <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
```

Comment:
I don't know why we have some of these references, but for now I'm just upgrading them.

This one might be related to our user-agent string.